compression_test.go

```diff
@@ -54,3 +56,29 @@ func TestAcceptEncodingOrdering(t *testing.T) {
 	_, _ = client.CallUnary(context.Background(), NewRequest(&emptypb.Empty{}))
 	assert.True(t, called)
 }
+
+type failDecompressor struct {
+	Decompressor
+}
+
+type failCompressor struct{}
+
+func (failCompressor) Write([]byte) (int, error) {
+	return 0, errors.New("failCompressor")
+}
+
+func (failCompressor) Close() error {
+	return errors.New("failCompressor")
+}
+
+func (failCompressor) Reset(io.Writer) {}
+
+func WithErrorCompression() Option {
```

Comment:
This is exported to be tested in `connect_ext_test.go`.

Reply:
I think we should be able to put this in `connect_ext_test.go` and unexport it:

```
func withErrorCompression() connect.Option {
  return connect.WithCompression(
    "error",
    func() connect.Decompressor { return &failDecompressor{} },
    func() connect.Compressor { return &failCompressor{} },
  )
}
```